### DISCUSSION TITLE
Cache expiry

### DISCUSSION THREAD

**Discussion by jaschaio** [SonicJs-Org/sonicjs#219] [Q&A] [answered]
I see in [the documentation how the cache works](https://sonicjs.com/persistence#caching-in-memory-and-kv-cache) (e.G. first in memory » KV » D1) but the one thing I wonder about is how does expiring the cache work?

e.G. to keep the example from the docs, if I request the URL `http://localhost:8788/v1/posts?limit=5&offset=2&sortBy=created_on&sortDirection=desc` once and afterwards add a new post, how does sonicjs know that the cache is expired and it should fetch new results from D1?

**Comment by lane711** [comment, marked as answer]
sorry for the late reponse - SonicJs automatically invalidates the cache when a record is updated/inserted or deleted
